python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1359,16 +1373,37 @@ def build_execution_context(
             " OpExecutionContext, or left blank."
         )
 
-    if context_annotation is EmptyAnnotation:
-        # if no type hint has been given, default to:
-        # * AssetExecutionContext for sda steps not in graph-backed assets, and asset_checks
-        # * OpExecutionContext for non sda steps
-        # * OpExecutionContext for ops in graph-backed assets
-        if is_asset_check:
-            return AssetExecutionContext(step_context)
-        if is_op_in_graph_asset or not is_sda_step:
-            return OpExecutionContext(step_context)
-        return AssetExecutionContext(step_context)
-    if context_annotation is AssetExecutionContext:
-        return AssetExecutionContext(step_context)
-    return OpExecutionContext(step_context)
+    asset_ctx = AssetExecutionContext(step_context)
+    op_ctx = OpExecutionContext(step_context)
```

Comment:
The thing that led me this current approach was ensuring that the passed down `context` was `==` to the result of `_ExecutionContext.get()`. It will requires some shenaigans to maintain that equality if we do the proposed shifting.

Reply:
I'm not sure I'm quite following the issue/proposed solution here. can you add some elaboration?